lblod/app-digitaal-loket
# [ Maintenance ] Consolidating leidinggevenden delta flow

# Description
DL-5582 -> DL-5583

This PR extends the configuration of the maintainer and job-initiator for Leidinggevenden deltas. It moves in config.json files environment variables of the docker-compose related to the services. It also removes the custom rule file to include it in the maintainer rule instead.

# Type of change


# Related services

- https://github.com/lblod/delta-producer-publication-graph-maintainer
- https://github.com/lblod/delta-producer-background-jobs-initiator
- https://github.com/lblod/delta-consumer-single-graph-maintainer (DEPRECATED)
- https://github.com/lblod/delta-consumer

# How to test 

1. Add your `key` and `value` in your `app-digitaal-loket/config/delta-producer/publication-graph-maintainer/config.json` file
```json
  "leidinggevenden": {
    "deltaPath": "/leidinggevenden/delta",
    "errorCreatorUri": "http://lblod.data.gift/services/delta-producer-publication-graph-maintainer-leidinggevenden",
    "exportConfigPath": "/config/leidinggevenden/export.json",
    "filesPath": "/leidinggevenden/files",
    "jobsGraph": "http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin",
    "relativeFilePath": "deltas/leidinggevenden",
    "healingJobOperation": "http://redpencil.data.gift/id/jobs/concept/JobOperation/deltas/healingOperation/leidinggevenden",
    "initialPublicationGraphSyncJobOperation": "http://redpencil.data.gift/id/jobs/concept/JobOperation/deltas/initialPublicationGraphSyncing/leidinggevenden",
    "queuePollInterval": 3000,
    "skipMuAuthInitialSync": true,
    "serveDeltaFiles": true,
    "publisherUri": "http://data.lblod.info/services/delta-production-json-diff-file-manager-leidinggevenden",
    "deltaInterval": 10000,
    "errorGraph": "http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin",
    "useVirtuosoForExpensiveSelects": true,
    "skipMuAuthHealing": true,
    "useFileDiff": true,
    "healingInitialBatchSizeInsert": 1000,
    "updatePublicationGraphSleep": 200,
    "publicationGraph": "http://redpencil.data.gift/id/deltas/producer/loket-leidinggevenden-producer",
    "filesGraph": "http://redpencil.data.gift/id/deltas/producer/leidinggevenden-deltas",
    "loginPath": "/leidinggevenden/login",
    "key" : "yourKey"
  }
```

2. drc up -d and log as any unit (ex : Gemeente Aalst)
3. In the Leidinggevenden module set a position for a person. You can monitor the logs of `delta-producer-publication-graph-maintainer` it will create a delta file once you click validate the step. This delta can then be consumed by [app-leidinggevenden-databank](https://github.com/lblod/app-leidinggevenden-databank). Note that it runs under this consumer https://github.com/lblod/delta-consumer-single-graph-maintainer and Docker compose variable are not the same as https://github.com/lblod/delta-consumer but you can update the consumer in the docker-compose-override.yml and use the generic delta-consumer
4. Depending of your delta-consumer service you'll need to set few variables 
```yml
  leidinggevenden-consumer:
    environment:
      DCR_DISABLE_INITIAL_SYNC: 'true'
      DCR_WAIT_FOR_INITIAL_SYNC: 'false'
      DCR_SECRET_KEY: "yourKey"
      DCR_SYNC_BASE_URL: 'http://endpoint-of-your-choice:90'
      DCR_SYNC_LOGIN_ENDPOINT: 'http://endpoint-of-your-choice:90/sync/leidinggevenden/login'
      DCR_START_FROM_DELTA_TIMESTAMP: "2024-01-09"
```
5. drc up -d then log the leidinggevenden-consumer, it should consume the delta file created from loket.

# What to check

- N/A

# Links to other PR's

- N/A

# Notes

Next step : Migrate to publication-triplestore
